cpp/src/gandiva/gdv_function_stubs.cc

```diff
@@ -794,6 +794,128 @@ const char* gdv_fn_initcap_utf8(int64_t context, const char* data, int32_t data_
   *out_len = out_idx;
   return out;
 }
+
+GANDIVA_EXPORT
+const char* gdv_fn_concat_ws_utf8_utf8(int64_t context, const char* separator,
+                                       int32_t separator_len, const char* word1,
+                                       int32_t word1_len, const char* word2,
+                                       int32_t word2_len, int32_t* out_len) {
+  if (word1_len < 0 && word2_len < 0) {
```

Comment:
If word1_len < 0 or word2_len < 0; the below code doesn't work as expected. Please change this to:

if (word1_len < 0 || word2_len < 0 || separator_len < 0) {
}

Reply:
Done